src/helpers/getResultsByResource.ts

```diff
@@ -0,0 +1,33 @@
+import omit from 'lodash/omit';
+import uniq from 'lodash/uniq';
+import { ChecksResults, FailedRule, ResourceResult } from '../types';
+
+export const getResultsByResource = (
+  results: ChecksResults,
+): ResourceResult[] => {
+  const checkedResources = uniq(
+    results.flatMap(({ result }) => result.map(({ arn }) => arn)),
+  );
+
+  const resultsByResource = checkedResources.map(resourceArn => {
+    const failedRules = results
+      .map(({ rule, result }) => {
```

Comment:
With a `flatMap`, you could remove `undefined` and avoid `filter`. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap#for_adding_and_removing_items_during_a_map